similar/arch/ogl/gr.cpp

```diff
@@ -1082,9 +1098,20 @@ void gr_palette_step_up(int r, int g, int b)
 	old_b_g = ogl_brightness_g;
 	old_b_b = ogl_brightness_b;
 
-	ogl_brightness_r = max(r + gr_palette_gamma, 0);
-	ogl_brightness_g = max(g + gr_palette_gamma, 0);
-	ogl_brightness_b = max(b + gr_palette_gamma, 0);
+	if (r >= 0)
```

Comment:
This block looks like it would benefit from an inline function to reduce the repetition.